internal/utils/version.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 Nethermind
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+import (
+	"context"
+	"errors"
+	"github.com/google/go-github/v47/github"
+	"os/exec"
+)
+
+const (
+	// owner is the owner of the repository
+	owner = "NethermindEth"
+	// repo is the name of the repository
+	repo = "sedge"
+	// MsgNeedVersionUpdate is the message to display when a new Version is available
+	MsgUnableToCheckVersion = "Unable to check for new Version. Please check manually at " +
+		"https://github.com/NethermindEth/sedge/releases, with error:"
+	MsgNeedVersionUpdate = "A new Version of sedge is available. Please update to the latest Version. See " +
+		"https://github.com/NethermindEth/sedge/releases for more information. Latest detected tag:"
+	MsgVersionUpdated = "You are running the latest version of sedge. Version: "
+)
+
+var ErrorNoTag = errors.New("no tag found on Github")
+var ErrorCheckingVersion = errors.New("error while checking for new Version, please check your internet connection")
+
+var Version string
+
+func init() {
+	if Version == "" {
+		version, err := versionFromGit()
+		if err != nil {
+			Version = "v0.0.0"
+		} else {
+			Version = version
+		}
+	}
+}
+
+func versionFromGit() (string, error) {
+	cmd := "git tag | sort | tail -n 1"
+	out, err := exec.Command("bash", "-c", cmd).Output()
```

Comment:
Please use the code and interfaces designed for running commands in the`internal/pkg/commands` package. Check examples in the `cli` command code to see how to use it. You would need to define a method for the`UnixCMDRunner` struct that executes the command you want. This is designed in a way is easier to maintain a cross platform code. As you probably have realized, this version check doesn't work on Windows right now.

Reply:
Please review new changes